WcaOnRails/app/javascript/edit-schedule/EditVenue/index.jsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react'
+import { rootRender } from 'edit-schedule'
+import {
+  convertVenueActivitiesToVenueTimezone,
+  newRoomId,
+  toMicrodegrees,
+  toDegrees,
+} from '../utils'
+import { EditRoom } from './EditRoom'
+import { compose, withProps } from "recompose"
+import { withGoogleMap, GoogleMap, Marker } from "react-google-maps"
+import { Panel, Row, Col, Clearfix } from 'react-bootstrap'
+import { timezoneData } from 'wca/timezoneData.js.erb'
+
+export class EditVenue extends React.Component {
+
+  handleSinglePropertyChange = (e, propName) => {
+    this.props.venueWcif[propName] = e.target.value;
+    if (propName == "timezone") {
+      convertVenueActivitiesToVenueTimezone(this.props.venueWcif);
+    }
+    rootRender();
+  }
+
+  markerRefUpdater = (input) => this.marker = input;
+
+  handlePositionChange = () => {
+    let pos = this.marker.getPosition();
```

Comment:
This feels like a bit of a weird pattern to me. Can you instead get rid of this `refUpdater` that you pass around everywhere and instead change this so `handlePositionChange` gets a `marker` as an argument?